biggerquery/user_commons/fastai/predict_component.py

```diff
@@ -0,0 +1,102 @@
+import runpy
+import uuid
+import logging
+from pathlib import Path
+from ...utils import unzip_file_and_save_outside_zip_as_tmp_file
+from . import predict_io
+
+
+def fastai_tabular_prediction_component(
+        input_table_name,
+        output_table_name,
+        partition_column,
+        model_file_path,
+        dataset,
+        torch_package_path,
+        fastai_package_path,
+        dataflow_job_name=None,
+        custom_input_collection=None,
+        custom_output=None,
+        custom_pipeline=None):
+
+    component = FastaiTabularPredictionComponent(
+        input_table_name,
+        output_table_name,
+        partition_column,
+        model_file_path,
+        dataset,
+        torch_package_path,
+        fastai_package_path,
+        dataflow_job_name,
+        custom_input_collection,
+        custom_output,
+        custom_pipeline
+    )
+
+    def component_callable(ds):
+        return component.run_component(ds.runtime_str)
+
+    return component_callable
+
+
+class FastaiTabularPredictionComponent(object):
+    def __init__(self,
+                 input_table_name,
+                 output_table_name,
+                 partition_column,
+                 model_file_path,
+                 dataset,
+                 torch_package_path,
+                 fastai_package_path,
+                 dataflow_job_name=None,
+                 custom_input_collection=None,
+                 custom_output=None,
+                 custom_pipeline=None):
+        self.input_table_name = input_table_name
+        self.output_table_name = output_table_name
+        self.partition_column = partition_column
+        self.model_file_path = model_file_path
+        self.config = dataset.config if dataset is not None else None
+        self.torch_package_path = torch_package_path
+        self.fastai_package_path = fastai_package_path
+        self.dataflow_job_name = dataflow_job_name or f'fastai-prediction-{uuid.uuid4()}'
+
+        self.custom_input_collection = custom_input_collection
+        self.custom_output = custom_output
+        self.custom_pipeline = custom_pipeline
+
+    def run_component(self, runtime):
+        logging.info('Running fastai prediction component')
+        runtime = runtime[:10]
```

Comment:
This is some magic number. I think it will be better if we create some method for it. maybe `getDate`

Reply:
I used named slice, done